### ISSUE TITLE
Update release documentation to only mention git

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8324]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8324|
|Type|task|
|Created|2016-05-10 16:12:26Z|

We're on git now, so we need to use it fully, incl doing our own tags and such.

Part of [#8322](https://github.com/twisted/twisted/issues/8322).

<details><summary>Searchable metadata</summary>

```
trac-id__8324 8324
type__task task
reporter__hawkowl hawkowl
priority__normal normal
milestone__None None
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__release_management release management
keywords__None None
time__1462896746473493 1462896746473493
changetime__1468435102085453 1468435102085453
version__None None
owner__None None
cc__radix
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

Will be fixed by [#8271](https://github.com/twisted/twisted/issues/8271)

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl set status to `closed`|
|-|-|

Fixed by [#8271](https://github.com/twisted/twisted/issues/8271)
